aws/table_aws_ec2_ami.go

```diff
@@ -349,3 +360,59 @@ func getEc2AmiTurbotTitle(_ context.Context, d *transform.TransformData) (interf
 
 	return title, nil
 }
+
+// // UTILITY FUNCTION
+// Build AMI's list call input filter
+func buildAmisWithOwnerFilter(input *ec2.DescribeImagesInput, quals plugin.KeyColumnQualMap, ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) []types.Filter {
+	filters := make([]types.Filter, 0)
+
+	filterQuals := map[string]string{
+		"architecture":        "architecture",
+		"description":         "description",
+		"ena_support":         "ena-support",
+		"hypervisor":          "hypervisor",
+		"image_type":          "image-type",
+		"kernel_id":           "kernel-id",
+		"name":                "name",
+		"platform":            "platform",
+		"public":              "is-public",
+		"ramdisk_id":          "ramdisk-id",
+		"root_device_name":    "root-device-name",
+		"root_device_type":    "root-device-type",
+		"state":               "state",
+		"sriov_net_support":   "sriov-net-support",
+		"virtualization_type": "virtualization-type",
+	}
+
+	columnsBool := []string{"ena_support", "public"}
+
+	for columnName, filterName := range filterQuals {
+		if quals[columnName] != nil {
+			filter := types.Filter{
+				Name: go_kit_pack.String(filterName),
+			}
+			if strings.Contains(fmt.Sprint(columnsBool), columnName) { //check Bool columns
+				value := getQualsValueByColumn(quals, columnName, "boolean")
+				filter.Values = []string{fmt.Sprint(value)}
+			} else {
+				value := getQualsValueByColumn(quals, columnName, "string")
+				val, ok := value.(string)
+				if ok {
+					filter.Values = []string{val}
+				}
+			}
+			filters = append(filters, filter)
+		}
+	}
+
+	// Use this section later and compare the results
```

Comment:
@bigdatasourav could we add a better comment as to why we are caching the data and what will be compared with?